File: r/src/list_strategies.R
library(httr2)
library(jsonlite)
library(dotenv)
library(glue)

# Load environment variables
load_dot_env()

BASE_URL <- "https://api.hyperswitchai.com"

# Source the cache_token.R file to get access to its functions
source("src/cache_token.R")

main <- function() {
  tryCatch({
    # Get cached token using the imported function
    token <- get_token()

    # List the strategies
    response <- request(glue("{BASE_URL}/admin/strategies/list")) %>%
      req_headers(
        "Content-Type" = "application/json",
        "Authorization" = glue("Bearer {token}")
      ) %>%
      req_method("GET") %>%
      req_perform()

    # Parse response
    result <- fromJSON(resp_body_string(response))
    
    # Replace NA with empty string for cleaner output
    result$strategies$endpoint[is.na(result$strategies$endpoint)] <- ""
    
    cat("\n\033[32m✅ Strategies listed successfully\033[0m\n\n")
    
    # Print strategies in a more readable format
    strategies <- result$strategies
    cat("Strategies:\n")
    for (i in 1:nrow(strategies)) {
      cat("\n", i, ".", 
          "\n   Code:", strategies$code[i],
          "\n   Type:", strategies$type[i],
          "\n   Endpoint:", strategies$endpoint[i], "\n")
    }

  }, error = function(e) {
    cat("\nError:", e$message, "\n")
    if (!is.null(e$body)) {
      cat("Response body:", e$body, "\n")
    }
  })
}

# Execute if run directly
if (sys.nframe() == 0) {
  main()
} 

File: ruby/src/list_strategies.rb
require 'dotenv'
require 'faraday'
require 'json'

# Load environment variables from .env file
Dotenv.load

BASE_URL = 'https://api.hyperswitchai.com'

# Source the cache_token.rb file to get access to its functions
require_relative 'cache_token'

def main
  begin
    # Get cached token using the imported function
    token = get_token

    # Create HTTP client
    conn = Faraday.new(url: BASE_URL) do |f|
      f.request :json
      f.response :json
    end

    # List all strategies
    response = conn.get('/admin/strategies/list') do |req|
      req.headers['Content-Type'] = 'application/json'
      req.headers['Authorization'] = "Bearer #{token}"
    end

    # Handle response
    unless response.success?
      puts "\n\e[31mServer response: #{response.status} #{response.body}\e[0m"
      raise "Failed to list strategies: #{response.reason_phrase}"
    end

    puts "\n\e[32m✅ Strategies listed successfully\e[0m"
    puts JSON.pretty_generate(response.body)

  rescue => e
    puts "Error: #{e.message}"
  end
end

# Run if this file is executed directly
main if __FILE__ == $PROGRAM_NAME 